modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java

```diff
@@ -319,34 +331,45 @@ public int execute(List<String> rawArgs) {
                     if (isSSLMisconfigurationError(cause))
                         e = cause;
 
-                    logger.severe("Connection to cluster failed. " + CommandLogger.errorMessage(e));
+                    logger.severe("Connection to cluster failed. " + errorMessage(e));
 
                 }
 
                 logger.info("Command [" + commandName + "] finished with code: " + EXIT_CODE_CONNECTION_FAILED);
 
+                err = e;
+
                 return EXIT_CODE_CONNECTION_FAILED;
             }
 
             if (X.hasCause(e, VisorIllegalStateException.class)) {
                 VisorIllegalStateException vise = X.cause(e, VisorIllegalStateException.class);
 
-                logger.severe(CommandLogger.errorMessage(vise));
+                logger.severe(errorMessage(vise));
                 logger.info("Command [" + commandName + "] finished with code: " + EXIT_CODE_ILLEGAL_STATE_ERROR);
 
+                err = e;
+
                 return EXIT_CODE_ILLEGAL_STATE_ERROR;
             }
 
-            logger.severe(CommandLogger.errorMessage(e));
+            logger.severe(errorMessage(e));
             logger.info("Command [" + commandName + "] finished with code: " + EXIT_CODE_UNEXPECTED_ERROR);
 
+            err = e;
+
             return EXIT_CODE_UNEXPECTED_ERROR;
         }
         finally {
             LocalDateTime endTime = LocalDateTime.now();
 
             Duration diff = Duration.between(startTime, endTime);
 
+            if (printErrStackTrace && nonNull(err)) {
+                logger.info("Error stack trace:");
+                logger.info(X.getFullStackTrace(err));
```

Comment:
Does this logger have an issue with printing several lines?
Like this manner:
logger.info("Error stack trace:\n" +
  X.getFullStackTrace(err));